src/main/java/de/tum/in/www1/artemis/service/connectors/jenkins/JenkinsService.java

```diff
@@ -508,7 +509,8 @@ private void addFeedbackToResult(Result result, TestResultsDTO report) {
             result.addFeedbacks(scaFeedback);
         }
 
-        result.setHasFeedback(!result.getFeedbacks().isEmpty());
+        // Relevant feedback is negative
+        result.setHasFeedback(result.getFeedbacks().stream().anyMatch(fb -> !fb.isPositive()));
```

Comment:
is it correct to set `hasFeedback` to false if all feedbacks are positive?

Reply:
BambooService and JenkinsService set this attribute differently before. The condition in the BambooService was
```
if (!job.getFailedTests().isEmpty() || hasStaticCodeAnalysisFeedback) {
                    result.setHasFeedback(true);
                }
```
Failed tests are negative and SCA feedback is also always negative. So the new condition is equivalent. I used the same condition for the JenkinsService now. Maybe we should move the setting of this to the GradingService.

Maybe this should be `result.setHasFeedback(!result.getFeedbacks().isEmpty());` depending on the semantics wanted for the attribute. @krusche can you clarify?

Reply:
The setting of this value here seems irrelevant as the GradingService overrides this later. We keep this now, and try to simplify it in another PR, where we can thoroughly test all programming exercise scenarios.